### PR TITLE
Update appnet interop

### PR DESCRIPTION
**Pull request by gnossen**
Changes for updates to API.

### CODE REVIEW

**Comment by gnossen** [comment]
[Passing run (manually triggered)](https://fusion2.corp.google.com/ci/guitar/projects/network_staging_us-west1-pj1-proxyless-grpc-app-net/workflows/%2F%2Fcloud%2Fcluster%2Ftesting%2Fguitar%2Fnetwork:network_staging_us-west1-pj1-proxyless-grpc-app-net/activity/3d424fdd-4a9d-3fdf-9e09-879089f94eee:0/tests)

**Comment by gnossen** [comment]
Runs of existing tests:
- [grpc/core/master/linux/psm-security](https://fusion2.corp.google.com/invocations/5642310c-4b0c-4d94-8c3a-d7f91d0019f5)
- [grpc/core/master/linux/grpc_xds_k8s_lb](https://fusion2.corp.google.com/invocations/d5559219-5ced-42c7-8cb3-957907d450f8)
- [grpc/core/master/linux/grpc_xds_url_map](https://fusion2.corp.google.com/invocations/1223a4a2-a172-448c-a7c5-528217e69112)

File: tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/client-secure.deployment.yaml [hunk 1]
```diff
@@ -63,8 +63,8 @@ spec:
             % if xds_server_uri:
             - "--xds-server-uri=${xds_server_uri}"
             % endif
-            % if config_scope:
-            - "--config-scope-experimental=${config_scope_experimental}"
+            % if config_mesh:
+            - "--config-mesh-experimental=${config_mesh_experimental}"
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/client-secure.deployment.yaml:67]:
```suggestion
            - "--config-mesh-experimental=${config_mesh}"
```

I think this was wrong in the first place, there isn't a variable `config_scope_experimental` set `framework.test_app.client_app.KubernetesClientRunner.run`. But we probably never ran security tests with the app net on.

File: tools/run_tests/xds_k8s_test_driver/tests/app_net_test.py [hunk 2]
```diff
@@ -35,7 +35,7 @@ def test_ping_pong(self):
             self.td.create_backend_service()
 
         with self.subTest('2_create_mesh'):
-            self.td.create_mesh()
+            mesh = self.td.create_mesh()
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/tests/app_net_test.py:38]: This variable is not used. Did you mean to use it on line 52, or this assignment is not needed?

File: tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py [hunk 3]
```diff
@@ -291,7 +289,8 @@ def run(self,
             qps=25,
             metadata='',
             secure_mode=False,
-            print_response=False) -> XdsTestClient:
+            print_response=False,
+            config_mesh=None) -> XdsTestClient:
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py:293]: nit: can we insert `config_mesh` between `secure_mode` and `print_response`? No real reason, just a nit - I was hoping `print_response` will always be last in the args, and inserted rows won't modify unrelated args.

File: tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py [hunk 4]
```diff
@@ -329,14 +328,14 @@ def run(self,
...
-            config_scope=self.config_scope,
             stats_port=self.stats_port,
             server_target=server_target,
             rpc=rpc,
             qps=qps,
             metadata=metadata,
             secure_mode=secure_mode,
-            print_response=print_response)
+            print_response=print_response,
+            config_mesh=config_mesh)
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py:338]: Same nit here. Totally optional.

**Review by sergiitk — APPROVED** [review]
LGTM. Minor suggestions, and a bug on a (currently) dead code path.

File: tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/network_services.py [hunk 5]
```diff
@@ -139,17 +133,13 @@ def from_response(cls, d: Dict[str, Any]) -> 'Destination':
...
         @classmethod
         def from_response(cls, d: Dict[str, Any]) -> 'RouteAction':
             destinations = [
                 Destination.from_response(dest) for dest in d["destinations"]
             ] if "destinations" in d else []
-            return cls(
-                destinations=destinations,
-                drop=d.get("drop"),
-            )
+            return cls(destinations=destinations,)
```
[the remark this answers is not in the document]
Comment (gnossen) [reply tools/run_tests/xds_k8s_test_driver/framework/infrastructure/gcp/network_services.py]: Done.

File: tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/client-secure.deployment.yaml (hunk 1 above)
Comment (gnossen) [reply tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/client-secure.deployment.yaml]: Ooh. Good catch. Fixed.

File: tools/run_tests/xds_k8s_test_driver/tests/app_net_test.py (hunk 2 above)
Comment (gnossen) [reply tools/run_tests/xds_k8s_test_driver/tests/app_net_test.py]: Remnant of a previous approach. This should have been cleaned up. Removed.

File: tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py (hunk 3 above)
Comment (gnossen) [reply tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py]: Done.

File: tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py (hunk 4 above)
Comment (gnossen) [reply tools/run_tests/xds_k8s_test_driver/framework/test_app/client_app.py]: Done.

**Comment by gnossen** [comment]
[Green re-run with review comments](https://fusion2.corp.google.com/ci/guitar/projects/network_staging_us-west1-pj1-proxyless-grpc-app-net/activity/ace14505-2a19-4bc5-a5d9-3acad3274a3d/overview)

If this causes any problems while I'm OOO for any reason, please feel free to revert.

**Pull request merged**